### PR TITLE
feat(PolyScript): add polygon script

### PR DESCRIPTION
**Pull request by Sotatek-Chung**

### CODE REVIEW

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+      .copyString("94")
+      .copyArgument(argTo)
+      // value
+      .rlpString(argValue)
+      // data
+      .copyString("80")
+      // accessList
+      .copyString("C0")
+      .arrayEnd(1)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:85]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+      .copyArgument(argContractAddress)
+      // data, Length = 68
+      .copyString("80B844a9059cbb000000000000000000000000")
+      .copyArgument(argTo)
+      .copyString("0000000000000000000000000000000000000000")
+      .copyArgument(argValue)
+      // accessList
+      .copyString("C0")
+      .arrayEnd(1)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:157]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+      .rlpString(argGasFeeCap)
+      .rlpString(argGasLimit)
+      .copyString("94")
+      .copyArgument(argTo)
+      .rlpString(argValue)
+      .rlpDataPlaceholder(argData)
+      // accessList
+      .copyString("C0")
+      .arrayEnd(1)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:219]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+      // data
+      .copyString("80")
+      // chainId v
+      // + ScriptAssembler.rlpString(argChainId)
+      .copyString("89", Buffer.CACHE1)
+      .rlpString(ScriptData.getDataBufferAll(Buffer.CACHE1))
+      // r,s
+      .copyString("8080")
+      .rlpList(1)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:276]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+      .copyString("0000000000000000000000000000000000000000")
+      .copyArgument(argValue)
+      // chainId v
+      //                .rlpString(argChainId)
+      .copyString("89", Buffer.CACHE1)
+      .rlpString(ScriptData.getDataBufferAll(Buffer.CACHE1))
+      // r,s
+      .copyString("8080")
+      .rlpList(2)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:346]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 6]
```diff
@@ -0,0 +1,462 @@
...
+      .rlpString(argValue)
+      // data
+      .rlpDataPlaceholder(argData)
+      // chainId v
+      //                 .rlpString(argChainId)
+      .copyString("89", Buffer.CACHE1)
+      .rlpString(ScriptData.getDataBufferAll(Buffer.CACHE1))
+      .copyString("8080")
+      .arrayEnd(1)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:412]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 7]
```diff
@@ -0,0 +1,462 @@
...
+  public static String getPolyMessageBlindScript() {
+    ScriptArgumentComposer sac = new ScriptArgumentComposer();
+    ScriptData argMessage = sac.getArgumentAll();
+
+    String script = new ScriptAssembler()
+      // set coinType to 3C
+      .setCoinType(0x3C)
+      .copyString("19457468657265756D205369676E6564204D6573736167653A0A")
+      .copyArgument(argMessage)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:432]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 8]
```diff
@@ -0,0 +1,462 @@
...
+    ScriptData argDomainSeparator = sac.getArgument(32);
+    ScriptData argMessage = sac.getArgumentAll();
+
+    String script = new ScriptAssembler()
+      // set coinType to 3C
+      .setCoinType(0x3C)
+      .copyString("1901")
+      .copyArgument(argDomainSeparator)
+      .hash(argMessage, Buffer.TRANSACTION, ScriptAssembler.Keccak256)
+      .showMessage("POLYGON")
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:454]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java (hunk 1 above)
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:85]: should display symbol "MATIC"

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 9]
```diff
@@ -0,0 +1,462 @@
...
+      .rlpString(argValue)
+      // data
+      .copyString("80")
+      // chainId v
+      // + ScriptAssembler.rlpString(argChainId)
+      .copyString("89", Buffer.CACHE1)
+      .rlpString(ScriptData.getDataBufferAll(Buffer.CACHE1))
+      // r,s
+      .copyString("8080")
+      .rlpList(1)
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:275]: rlpList is deprecated function

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 10]
```diff
@@ -0,0 +1,462 @@
...
+      .copyArgument(argTo)
+      .copyString("0000000000000000000000000000000000000000")
+      .copyArgument(argValue)
+      // chainId v
+      //                .rlpString(argChainId)
+      .copyString("89", Buffer.CACHE1)
+      .rlpString(ScriptData.getDataBufferAll(Buffer.CACHE1))
+      // r,s
+      .copyString("8080")
+      .rlpList(2)
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:345]: rlpList is deprecated function

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 11]
```diff
@@ -0,0 +1,462 @@
...
+80
+     */
+  public static String getERC20Script() {
+    ScriptArgumentComposer sac = new ScriptArgumentComposer();
+    ScriptData argTo = sac.getArgument(20);
+    ScriptData argValue = sac.getArgument(12);
+    ScriptData argGasPrice = sac.getArgumentRightJustified(10);
+    ScriptData argGasLimit = sac.getArgumentRightJustified(10);
+    ScriptData argNonce = sac.getArgumentRightJustified(8);
+    ScriptData argChainId = sac.getArgumentRightJustified(2);
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:317]: redundant argument

**Review by hankliu418 — CHANGES_REQUESTED** [review]
Please update the issue below, thanks.

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java [hunk 12]
```diff
@@ -0,0 +1,423 @@
...
+      .copyArgument(argDomainSeparator)
+      .hash(argMessage, Buffer.TRANSACTION, ScriptAssembler.Keccak256)
+      .showMessage("MATIC")
+      .showWrap("EIP712", "")
+      .showPressButton()
+      //version=00 ScriptAssembler.hash=06=ScriptAssembler.Keccak256 sign=01=ECDSA
+      .setHeader(HashType.Keccak256, SignType.ECDSA)
+      .getScript();
+    return script;
+  }
```
Comment (hankliu418) [inline script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java:423]: missing end "}"

**Review by hankliu418 — CHANGES_REQUESTED** [review]
Please remove redundant argument in line 320, thanks.

File: script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java (hunk 11 above)
Comment (Sotatek-Chung) [reply script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/PolyScript.java]: we removed it.

**Review by hank121314 — APPROVED** [review]

**Review by srt1678 — APPROVED** [review]

**Pull request merged**